R/summary_functions.R

```diff
@@ -4,8 +4,9 @@
 #' @export
 percent_coverage <- function(coverage_result){
   coverage_data <- as.data.frame(coverage_result)
-  num_lines <- nrow(coverage_data)
-  num_not_zero <- sum(coverage_data$value != 0) # how many lines have a value that is not zero?
+  num_lines <- length(unique(coverage_data$first_line))
```

Comment:
I guess this depends on what you interpret percent_coverage to mean.  The
way I interpret it is percent of all traced expressions covered, so two
expressions on the same line are counted separately.  You seem to consider
percent_coverage as the percentage of lines covered, in which case your fix
works, but it was not my original intent.

This is ultimately my responsibility as I did not closely review the logic
for the pull request which introduced the percent_coverage function, I
should have made the distinction more clear.

On Tue, Jan 13, 2015 at 9:55 AM, Karl Forner <notifications@github.com>
wrote:

> In R/summary_functions.R
> <https://github.com/jimhester/covr/pull/30#discussion-diff-22866299>:
>
> > @@ -4,8 +4,9 @@
> >  #' @export
> >  percent_coverage <- function(coverage_result){
> >    coverage_data <- as.data.frame(coverage_result)
> > -  num_lines <- nrow(coverage_data)
> > -  num_not_zero <- sum(coverage_data$value != 0) # how many lines have a value that is not zero?
> > +  num_lines <- length(unique(coverage_data$first_line))
>
> As I wrote in the introduction, the sum(coverage_result$value > 0) /
> NROW(coverage_result) is wrong because it assumes that all first_line are
> unique. I provided a test that shows it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jimhester/covr/pull/30/files#r22866299>.
>